### PR TITLE
Clean up some stat series conversion, filter logic

### PR DESCRIPTION
**Pull request by shifucun**
The ObsSeries struct is used for the old GetStats() API and has a hack for reusing fields. Split it into two structs for clean logic. This is to prepare for merging time series cache data.

Also updated prod recon config based on most recent deployment.

### CODE REVIEW

File: deploy/gke/recon-prod.yaml [hunk 1]
```diff
@@ -15,10 +15,10 @@
 # Production Recon Service GKE params
 
 project: datcom-recon-prod
-ip: 34.98.98.180
+ip: 34.117.28.86
 domain: recon.datacommons.org
 api_title: Data Commons Recon (Prod)
 api: datacommons.Recon
-region: us-central1
+region: us-west1
```
Comment (shifucun) [inline deploy/gke/recon-prod.yaml:22]: There is still something issue with this deployment, and this config is for testing. Going to switch it back to us-central1 when the issue is solved.

**Review by beets — APPROVED** [review]
thanks for bringing more sanity to our stat protos :) 

just to confirm that these changes are internal and there's no change to api responses?

also, i don't understand the change in the recon test. is that more related to the config change? in general, it'd be best if these config updates were done in separate pr's for easy rollback, debugging, ...

**Comment by shifucun** [comment]
> thanks for bringing more sanity to our stat protos :)
> 
> just to confirm that these changes are internal and there's no change to api responses?
> 
> also, i don't understand the change in the recon test. is that more related to the config change? in general, it'd be best if these config updates were done in separate pr's for easy rollback, debugging, ...

Yea, this does not change the response, as reflected in golden tests.

Looks like the recon tests result is not stable.. need to spend some time or ask Wei to look into that. For the config change, it's still WIP, this is to match the current deployment..

**Comment by shifucun** [comment]
Thanks for the review!

File: internal/server/recon/golden/resolve_entities/result.json [hunk 2]
```diff
@@ -1,43 +1,43 @@
 {
   "resolvedEntities": [
     {
-      "sourceId": "newId/SunnyvaleId",
+      "sourceId": "newId/VietnamId",
```
Comment (shifucun) [inline internal/server/recon/golden/resolve_entities/result.json:4]: @Spaceenter can you check if the response is stable or if we need to do some sorting here?

Comment (pradh) [reply internal/server/recon/golden/resolve_entities/result.json:4]: Looks it is sorted - https://github.com/datacommonsorg/mixer/blob/master/internal/server/recon/resolve.go#L209-L212 .. I think the new order is right?

**Pull request merged**